client/lib/purchases/index.ts

```diff
@@ -130,7 +130,7 @@ export function getDisplayName( purchase: Purchase ): ReactChild | string | unde
 
 export function getPartnerName( purchase: Purchase ): string | null {
 	if ( isPartnerPurchase( purchase ) ) {
-		return purchase.partnerName;
+		return purchase.partnerName ?? null;
```

Comment:
This provides a null fallback instead of undefined as the function type requires.